src/ophyd_async/core/_device_filler.py

```diff
@@ -150,7 +156,7 @@ def create_signals_from_annotations(
             yield backend, extras
             signal = child_type(backend)
             for anno in extras:
-                assert isinstance(anno, DeviceAnnotation), anno
+                self.check_device_annotation(annotation=anno)
                 anno(self._device, signal)
```

Comment:
Then this would be:
```suggestion
                device_annotation = _check_device_annotation(anno)
                device_annotation(self._device, signal)
```

Reply:
done de964ad5bb209ef14ed5c5452eb0645de7b7c61b